lib/screens/wallet/components/transaction_details_bottom_sheet.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:intl/intl.dart';
+import 'package:seeds/components/profile_avatar.dart';
+import 'package:seeds/constants/app_colors.dart';
+import 'package:seeds/datasource/local/settings_storage.dart';
+import 'package:seeds/datasource/remote/model/transaction_model.dart';
+import 'package:seeds/design/app_theme.dart';
+import 'package:seeds/domain-shared/page_state.dart';
+import 'package:seeds/screens/wallet/interactor/viewmodels/member_bloc.dart';
+import 'package:seeds/utils/string_extension.dart';
+import 'package:share/share.dart';
+
+class TransactionDetailsBottomSheet extends StatelessWidget {
+  final TransactionModel transaction;
+
+  const TransactionDetailsBottomSheet(this.transaction, {Key? key}) : super(key: key);
+
+  void show(BuildContext context) {
+    showModalBottomSheet(
+      shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.only(topLeft: Radius.circular(40.0), topRight: Radius.circular(40.0)),
+      ),
+      context: context,
+      builder: (_) => this,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final account = settingsStorage.accountName;
+    return BlocProvider(
+      create: (_) =>
+          MemberBloc(transaction.to == account ? transaction.from : transaction.to)..add(const OnLoadMemberData()),
+      child: BlocBuilder<MemberBloc, MemberState>(
+        builder: (context, state) {
+          switch (state.pageState) {
+            case PageState.loading:
+              return const Padding(
+                padding: EdgeInsets.symmetric(vertical: 20.0),
+                child: Center(child: CircularProgressIndicator()),
+              );
+            case PageState.success:
+              return Stack(
+                children: [
+                  if (state.pageState == PageState.loading)
+                    Positioned.fill(
+                      child: Container(
+                        color: AppColors.primary,
+                        child: const Align(child: CircularProgressIndicator()),
+                      ),
+                    ),
+                  Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      const SizedBox(height: 12.0),
+                      Text(
+                        DateFormat("EEE dd MMM y kk:mm:ss").format(transaction.timestamp),
```

Comment:
maybe move this away from the UI?

Reply:
mmm Not quite sure Its just text format and just one line logic What do you think @gguijarro-c-chwy